vowpalwabbit/fb_parser/src/parse_example_flatbuffer.cc

```diff
@@ -271,8 +368,36 @@ bool get_namespace_hash(VW::workspace* all, const Namespace* ns, uint64_t& hash)
   return false;
 }
 
+bool features_have_names(const Namespace& ns)
+{
+  return flatbuffers::IsFieldPresent(&ns, Namespace::VT_FEATURE_NAMES) && (ns.feature_names()->size() != 0);
+  // TODO: It is not clear what the right answer is when feature_values->size is 0
+}
+
+bool features_have_hashes(const Namespace& ns)
+{
+  return flatbuffers::IsFieldPresent(&ns, Namespace::VT_FEATURE_HASHES) && (ns.feature_hashes()->size() != 0);
+}
+
+bool features_have_values(const Namespace& ns)
+{
+  return flatbuffers::IsFieldPresent(&ns, Namespace::VT_FEATURE_VALUES) && (ns.feature_values()->size() != 0);
+}
+
 int parser::parse_namespaces(VW::workspace* all, example* ae, const Namespace* ns, VW::experimental::api_status* status)
 {
+#define RETURN_NS_PARSER_ERROR(status, error_code)                                                                 \
+  if (_active_collection && _active_multi_ex)                                                                      \
```

Comment:
I don't quite understand what is happening in this state.  A comment would be helpful.